src/useFormo.ts

```diff
@@ -362,22 +373,19 @@ export function useFormo<
 
   const [submissionCount, setSubmissionCount] = useState(0);
 
-  function isArrayValue(v: unknown): v is Record<string, unknown>[] {
-    // NOTE(gabro): this is a loose check, we are not really checking all
-    // elements are Record <string, unknown> but we statically enforce that using
-    // the types
-    return Array.isArray(v);
-  }
+  const fileArrayKeys = Object.keys(initialValues).filter(
+    (k) => initialValues[k] && (initialValues[k] as any).__isFieldArray
+  );
```

Comment:
these are stored initially, based on initialValues, since keeping this logic stateless based on current values would break as soon as the values are updated by internal logic (the `__isFieldArray` is added on initial values only)